### PR TITLE
Add customization of crate names for external libraries

### PR DESCRIPTION
**Pull request by EPashkin**
Added extra config for external libraries
```
[options]
external_libraries = [
   "GLib",
   "Gdk",
   "GdkPixbuf",
]

[external_libraries]
crate-name="NameSpace"
```

Edit: switched parameters

### CODE REVIEW

**Comment by EPashkin** [comment]
@antoyo This allow fix [problem](https://github.com/antoyo/webkit2gtk-rs/issues/4#issuecomment-310909823), check it please.
@sdroege, maybe it fixed your problem too.

**Comment by antoyo** [comment]
It works fine.
Thanks!

**Comment by EPashkin** [comment]
Seems I take wrong turn.
May be it better be this?
```
[external_libraries]
crate-name="NameSpace"
```

**Comment by antoyo** [comment]
I don't mind: either way is fine.

**Comment by EPashkin** [comment]
I better change

**Comment by EPashkin** [comment]
Updated

File: src/config/external_libraries.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+   "GLib",
+   "Gdk",
+   "GdkPixbuf",
+]
+
+[external_libraries]
+coollib="CoolLib"
+other-lib="OtherLib"
+"#,
+        );
```
Comment (EPashkin) [inline src/config/external_libraries.rs]: Very strange format for function call (with ending comma), but as it compiles seems it OK.

File: src/config/external_libraries.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+[options]
+external_libraries = [
+   "GLib",
+   "Gdk",
+   "GdkPixbuf",
+]
+
+[external_libraries]
+coollib="CoolLib"
+other-lib="OtherLib"
```
Comment (sdroege) [inline src/config/external_libraries.rs]: Are all possible crate names valid toml keys? Or are there crate names that wouldn't be allowed here?

Comment (EPashkin) [inline src/config/external_libraries.rs]: If it valid for `[dependencies]`

Comment (sdroege) [inline src/config/external_libraries.rs]: Good point! 👍

**Comment by EPashkin** [comment]
@sdroege, this enough for GStreamer-sys?

**Comment by sdroege** [comment]
This doesn't allow renaming the sys crate itself, right? Only other sys crates it is depending on?

**Comment by EPashkin** [comment]
Only dependencies, for current crate name better fix https://github.com/gtk-rs/gir/blob/master/src/nameutil.rs#L136. But IMHO it correctly works for `GStreamer`:
`assert_eq!(crate_name("GStreamer"), "gstreamer");`
Can you say more what you want?

**Comment by sdroege** [comment]
For the -sys crate, the generated crate name is `gst` (and `gst-base` etc). I want it to be `gstreamer` (and `gstreamer-base` etc).

**Comment by EPashkin** [comment]
It strange: Gir don't change `name` in `[lib]` and `[package]` in `Cargo.toml` on regen,
so change need only once. Seems I missed something.

**Comment by EPashkin** [comment]
@sdroege, Tried regen for https://github.com/sdroege/gstreamer-sys,
other that now customized external libraries I found only problematic rewriting `package.links` with crate name,
so simply moved it to part, that fill empty Cargo.toml

**Comment by sdroege** [comment]
Fine with me if that all works :) Thanks!

**Comment by sdroege** [comment]
@EPashkin Can you send me the changes you did for gstreamer-sys so I can merge them?

**Comment by EPashkin** [comment]
Sorry, I don't make any changes in gstreamer-sys. Just checked what changes on regen.

**Comment by sdroege** [comment]
Ah, no problem then :) Just wanted to prevent doing your changes another time

**Pull request merged**